apis/v1alpha2/validation/httproute.go

```diff
@@ -166,3 +166,31 @@ func validateHTTPPathMatch(path *gatewayv1a2.HTTPPathMatch, fldPath *field.Path)
 	}
 	return allErrs
 }
+
+// validateHTTPRouteFilterTypeMatchesValues validates that the filter type matches the filter value.
+func validateHTTPRouteFilterTypeMatchesValues(filter *gatewayv1a2.HTTPRouteFilter, path *field.Path) field.ErrorList {
```

Comment:
It doesn't look like this is actually being called anywhere other than the test. I think the best place to fit this in would be in `validateHTTPRouteUniqueFilters`, with that func renamed to `validateHTTPRouteFilters`

Reply:
Would `validateHTTPRouteSpec` be a better fit to add the test? I noticed that `validateHTTPRouteUniqueFilters` is being called there along with `validateHTTPRouteBackendServicePorts`.

Reply:
Yeah not sure where the best place is, something needs to get called in `validateHTTPRouteSpec`. My suggestion would be to make that a generic function that validated a list of filters, including both uniqueness and that required fields were non-empty. That would also lead to another area for improvement, we would no longer need separate `validateHTTPRouteUniqueFilters` and `validateHTTPBackendUniqueFilters` functions.

Reply:
I did some refactoring. Please let me know what you think.